src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -54,9 +54,18 @@ public Transaction BuildTransaction(TransactionBuildContext context)
             if (context.Shuffle)
                 context.TransactionBuilder.Shuffle();
 
-            Transaction transaction = context.TransactionBuilder.BuildTransaction(context.Sign);
+            bool validationSuccess = false;
+            Transaction transaction = null;
+            TransactionPolicyError[] errors = null;
 
-            if (context.TransactionBuilder.Verify(transaction, out TransactionPolicyError[] errors))
+            var retryOptions = new RetryOptions(3, TimeSpan.FromMilliseconds(5), RetryStrategyType.Simple, typeof(ArgumentNullException));
+            RetryStrategy.Run(retryOptions, () =>
+            {
+                transaction = context.TransactionBuilder.BuildTransaction(context.Sign);
+                validationSuccess = context.TransactionBuilder.Verify(transaction, out errors);
+            });
+
+            if (validationSuccess)
```

Comment:
so we retry without checking the error reason?
I think we should retry only if errors are "volatile" and a rerun may effectively generate a valid transaction

Reply:
We will fix this 👍